src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,88 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.exception.MissingArgumentException;
+import duke.util.Parser;
+
+/**
+ * The Deadline Class represents a task that needs to be done before a specific date/time.
+ *
+ * It contains information relating to the task:
+ * - description
+ * - isDone
+ * - dueDate
+ * - dueTime
+ *
+ * @author Benedict Chua
+ */
+public class Deadline extends Task {
+    private LocalDate dueDate;
+    private LocalTime dueTime;
+
+    /**
+     * Constructs a new Deadline task from the given description and dueDate.
+     *
+     * @param description String of the task description.
+     * @param dueDate String of the given due date and time.
+     */
+    public Deadline(String description, String dueDate) {
+        super(description);
+
+        String[] dateInfo = dueDate.split(" ", 2);
+        if (dateInfo.length < 2) {
+            throw new MissingArgumentException("Date or Time", "Deadline");
+        }
+        this.dueDate = Parser.parseDate(dateInfo[0]);
+        this.dueTime = Parser.parseTime(dateInfo[1]);
+    }
+
+    /**
+     * Constructs a Deadline task from an existing task description, dueDate and completion status.
+     * Used when loading from a save file.
+     *
+     * @param completed String indicating the status of completion: 1 if done, 0 if not.
+     * @param description String of the task description.
+     * @param dueDate String of the given due date and time.
+     */
+    public Deadline(String completed, String description, String dueDate) {
```

Comment:
Naming `completed` variable to `completeStatus` to match the intention more closely

Reply:
Good catch! I think it might be even better if I make it into an enumeration now that I noticed this again.